canon/src/repr_host.rs

```diff
@@ -80,22 +78,53 @@ where
                 ident,
                 store: source.store().clone(),
             })
+        } else {
+            // inlined value
+            Ok(Repr::Value {
+                rc: Rc::new(T::read(source)?),
+                cached_ident: RefCell::new(None),
+            })
         }
     }
 
     fn encoded_len(&self) -> usize {
         let ident_len = S::Ident::default().as_ref().len();
+
         match &self {
             Repr::Value { rc, .. } => {
-                // If the length is larger nthan `ident_len` + 1,
+                // If the length is larger than `ident_len` + 1,
```

Comment:
Just a nit, shouldn't this comment be 'smaller than'?

Reply:
comment is still correct, but i clarified it a bit

Reply:
reworded this